draft-ietf-masque-connect-udp.md

```diff
@@ -403,18 +442,90 @@ Reference:
 : This document.
 
 
-## Datagram Format Type {#iana-format-type}
+--- back
+
+# Example Extensions
 
-This document will request IANA to register UDP_PAYLOAD in the "HTTP Datagram
-Format Types" registry established by {{HTTP-DGRAM}}.
+Extensions can define new semantics for the payload of HTTP Datagrams. The
+extension can then have an endpoint pick an available context ID using the
+allocation service ({{context-id}}) and register that context ID with their
+peer.
 
-|    Type     |   Value   | Specification |
-|:------------|:----------|:--------------|
-| UDP_PAYLOAD | 0xff6f00  | This Document |
-{: #iana-format-type-table title="Registered Datagram Format Type"}
 
+## Registering Contexts with Headers
+
+Extensions can define a new HTTP header to register a context ID with the peer
+endpoint.
+
+As an example, take an extension that conveys the time at which a UDP
+packet was received. The extension would first define the format of its HTTP
+Datagram Payload field:
+
+~~~
+UDP with Timestamp HTTP Datagrams {
+  Context ID (i),
+  Timestamp (64),
+  UDP Payload (..),
+}
+~~~
+{: #ex-dgram title="Example: Format of UDP Payload with Timestamp"}
+
+The extension would also define a new HTTP header (Example-UDP-Timestamps) that
+includes a context ID value. Servers that understand this new HTTP header would
+be able to consequently handle and parse datagrams on the context ID, while all
+other servers would silently drop the datagrams. Clients also indicate their
+support for receiving on this Context ID by registering it.
+
+~~~
+HEADERS
+:method = CONNECT
+:protocol = connect-udp
+:scheme = https
+:path = /192.0.2.42/443/
+:authority = proxy.example.org
+example-udp-timestamps = 42
+~~~
+{: #ex-hdr title="Example: Registration via header"}
+
+In this example request, HTTP Datagrams with context ID zero would only contain
+the UDP payload, whereas HTTP Datagrams with context ID 42 would also contain a
+timestamp.
+
+
+## Registering Contexts with Capsules
+
+Extensions can define a new Capsule type (see {{HTTP-DGRAM}}) to register a
+context ID with the peer endpoint.
+
+As an example, take an extension that compresses QUIC Connection IDs when the
+client is running QUIC over a UDP proxying tunnel. The extension would first
+define the transform applied to UDP payloads when compressing and decompressing,
+such as removing the bytes of the connection ID.
+
+The extension would also define a new capsule type
+(EXAMPLE_REGISTER_COMPRESSED_QUIC_CID) that includes a context ID value and the
+connection ID to compress. Endpoints that understand this new capsule type would
+be able to consequently handle and parse datagrams on the context ID, while all
+other endpoints would ignore the datagrams.
```

Comment:
Again, I see that this is an example, but I don't see how this works as a practical matter. What acknowledges this? How do I discover that the peer doesn't support this extension? Or does, for that matter? I can't just register a compressed CID and then wait to see if my  data goes through.

Reply:
Added a sentence about needing a header